File: lab3/FileReverse.c
//FileReverse.c
//Aaron Hom
//awhom
//lab3
//Takes an input file and reverse each token on a separate line

/*
*  FileIO.c
*  Reads input file and prints each word on a separate line of
*  the output file.
*/

#include<stdio.h>
#include<stdlib.h>
#include<string.h>

//reverses the array "word" (based off of PA1's reverseArray3())
void stringReverse(char* word){
    int a = 0;
    int temp = 0; 
    int b = strlen(word)-1;
    while(a < b){
	temp = word[a];
        word[a] = word[b];
        word[b] = temp;
        a++;
        b--;
    }
}

int main(int argc, char* argv[]){

FILE* in; /* file handle for input */ 
FILE* out; /* file handle for output*/
char word[256]; /* char array to store words from input file */

/* check command line for correct number of arguments */
if( argc != 3 ){
    printf("Usage: %s <input file> <output file>\n", argv[0]);
    exit(EXIT_FAILURE);
}

/* open input file for reading */
in = fopen(argv[1], "r");
if( in==NULL ){
    printf("Unable to read from file %s\n", argv[1]);
    exit(EXIT_FAILURE);
}

/* open output file for writing */
out = fopen(argv[2], "w");
if( out==NULL ){
    printf("Unable to write to file %s\n", argv[2]);
    exit(EXIT_FAILURE);
}

/* read words from input file, print on separate lines to output file*/
while(fscanf(in, " %s", word) !=EOF){
    stringReverse(word);
    fprintf(out, "%s\n", word);
}

/* close input and output files */
fclose(in);
fclose(out);
return(EXIT_SUCCESS);
}

File: lab2/FileReverse.java
//FileReverse.java
//Aaron Hom
//awhom
//lab2
//Takes an input file, reverses the contents and prints result

import java.io.*;
import java.util.Scanner;

class FileReverse{
    public static void main(String[] args) throws IOException{
    Scanner in = null;
    PrintWriter out = null;
    String line = null;
    String[] token = null;
    int i, n, lineNumber = 0;

// check number of command line arguments is at least 2
    if(args.length < 2){
        System.out.println("Usage: FileCopy <input file> <output file>");
        System.exit(1);
    }

// open files
    in = new Scanner(new File(args[0]));
    out = new PrintWriter(new FileWriter(args[1]));

// read lines from in, extract and print tokens from each line
    while( in.hasNextLine() ){
        lineNumber++;

// trim leading and trailing spaces, then add one trailing space so 
// split works on blank lines
        line = in.nextLine().trim() + " "; 
// split line around white space 
        token = line.split("\\s+");  
// print out tokens       
        n = token.length;
        for(i=0; i<n; i++){
            out.println(stringReverse(token[i], token[i].length()));
        }
    }

// close files
in.close();
out.close();
    }

    public static String stringReverse(String s, int n){
	String t = "";
        if (s.length() <=  1){        
            return s;
        }
        if (n > 0){         
	    t = s.charAt(n-1) + stringReverse(s, n-1);
        }
        return t;
        }
    }
